src/components/post-list/Posts.tsx

```diff
@@ -1,31 +1,75 @@
 import defaultPicture from "../../assets/images/defaultProfile.jpeg";
 import defaultPostPicture from "../../assets/images/defaultPostImage.jpeg";
-
+import { useAuth } from "../../context/AuthContext";
+import { useNavigate } from "react-router-dom";
 
 const Posts = (props: any) => {
 
   const { users, posts } = props;
 
+  const { currentUser }: any = useAuth();
+  const navigate = useNavigate();
+
   users.map((user: { listOfPosts: Array<Object>; id: string }) => {
-    return user.listOfPosts = posts.filter(((post: { userId: string; }) => post.userId === user.id));
+    return user.listOfPosts = posts.filter(((post: { userId: string; }) => post.userId === user.id)); 
   })
+  
+
+  let userPosts: any[] = [];
+
+  users.map((user: { listOfPosts: Array<Object>, id: string }) => {
+    user.listOfPosts.map((post: any) => {
+      return userPosts.push(post);
+    });
+  })
+
+  // Sorts posts by date created (newest posts at top)
+  let newResults = userPosts.sort((a, b) => {
+    return b.dateCreated - a.dateCreated;
+  });
+
 
-  const unixConversion = (dt: number) => {
+  // Converts to time and date for each post
```

Comment:
Not a biggy but move this to utils, not yet but I am sure dates will be needed plenty of other places

Reply:
Yeah good idea